tests/helpers/warning-handler.js

```diff
@@ -0,0 +1,22 @@
+import { registerWarnHandler } from '@ember/debug';
+
+let errorOnInternalWarnings = true;
+export let capturedWarnings = null;
+
+export function captureWarnings() {
+  capturedWarnings = [];
+  errorOnInternalWarnings = false;
+}
+
+registerWarnHandler(function (message, options, next) {
+  if (options.id.includes('ember-m3')) {
+    if (errorOnInternalWarnings) {
+      throw new Error(`Unhandled Warning: ${message}`);
+    }
+
+    // gather warnings for later assertion
+    capturedWarnings.push([message, options]);
+  } else {
+    next(message, options);
+  }
+});
```

Comment:
seems weird that there's no inverse of `captureWarnings`?  should  we have a `reset`?

As it stands whether a warning in a test that never invokes `captureWarnings` throws an error on warning depends on test-ordering (i.e. is there a prior test that *did* invoke `captureWarnings`).

Reply:
Good catch! Definitely a mistake.